--- a/CMakeLists.txt	2021-09-07 16:36:53.002660070 +0200
+++ b/CMakeLists.txt	2021-09-07 16:37:39.359384754 +0200
@@ -170,8 +170,8 @@
 )
 source_group(scripts FILES ${SQLITECPP_SCRIPT})
 
-# add sources of the wrapper as a "SQLiteCpp" static library
-add_library(SQLiteCpp ${SQLITECPP_SRC} ${SQLITECPP_INC} ${SQLITECPP_DOC} ${SQLITECPP_SCRIPT})
+# add sources of the wrapper as a "SQLiteCpp" dynamic library
+add_library(SQLiteCpp SHARED ${SQLITECPP_SRC} ${SQLITECPP_INC} ${SQLITECPP_DOC} ${SQLITECPP_SCRIPT})
 
 # Options relative to SQLite and SQLiteC++ functions
 
