--- mini_httpd-1.24/Makefile
+++ mini_httpd-1.24/Makefile
@@ -22,7 +22,7 @@ CRYPT_LIB =    -lcrypt
 
 BINDIR =	/usr/sbin
 MANDIR =	/usr/share/man
-CC =		cc
+#CC =		gcc
 CDEFS =		$(SSL_DEFS) $(SSL_INC)
 CFLAGS =	-O $(CDEFS) -ansi -pedantic -U__STRICT_ANSI__ -Wall -Wpointer-arith -Wshadow -Wcast-qual -Wcast-align -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wno-long-long
 LDFLAGS =	-s
