commit 104d3b6725f5071333b79bdcffb0f983ecf3474d
Author: Louis Berryer <louis.berryer@legrand.fr>
Date:   Mon Aug 24 14:42:42 2015 +0200

    Removing debug compiler options

diff --git a/Makefile.linux-nocmake b/Makefile.linux-nocmake
index acac5f2..1f6b2a5 100644
--- a/Makefile.linux-nocmake
+++ b/Makefile.linux-nocmake
@@ -1,4 +1,4 @@
-CXXFLAGS = -g -O0 -Wall -fPIC -pedantic -Iinclude
+CXXFLAGS = -O3 -Wall -fPIC -pedantic -Iinclude
 LDFLAGS = 
 
 EXEC = libserialcpp.so
